framework/test/jest/unit/specs/modules/chain/synchronizer/block_synchronization_mechanism/block_synchronization_mechanism.spec.js

```diff
@@ -14,130 +14,238 @@
 
 'use strict';
 
-const BlockSynchronizationMechanism = require('../../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
+const {
+	BlockSynchronizationMechanism,
+} = require('../../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
 const peersList = require('./peers');
+const {
+	Block: blockFixture,
+} = require('../../../../../../../mocha/fixtures/blocks');
 
-describe('BlockSynchronizationMechanism', () => {
-	const stubs = {
-		channel: {
-			invoke: jest.fn(),
-		},
-		modules: {
-			blocks: {
-				lastBlock: jest.fn(),
-			},
-		},
-	};
-	let blockSynchronizationMechanism;
-
-	beforeEach(() => {
-		blockSynchronizationMechanism = new BlockSynchronizationMechanism({
-			channel: stubs.channel,
-			modules: stubs.modules,
-		});
+describe('block_synchronization_mechanism', () => {
+	afterEach(async () => {
+		jest.clearAllMocks();
 	});
 
-	describe('#constructor', () => {});
+	describe('BlockSynchronizationMechanism', () => {
+		const activeDelegates = 101;
+		const channelMock = { invoke: jest.fn() };
 
-	describe('#_computeBestPeer', () => {
-		it('should accept an array of peers as input', () => {
-			expect(blockSynchronizationMechanism._computeBestPeer).toHaveLength(1);
+		const lastBlockGetterMock = jest.fn();
+		const blocksMock = {};
+		Object.defineProperty(blocksMock, 'lastBlock', {
+			get: lastBlockGetterMock,
 		});
 
-		it('should return a peer object', () => {
-			stubs.modules.blocks.lastBlock = {
-				height: 0,
-				prevotedConfirmedUptoHeight: 0,
-			};
-
-			const peers = [
-				{
-					lastBlockId: '12343245',
-					prevotedConfirmedUptoHeight: 1,
-					height: 67,
-					ip: '127.0.0.1',
+		const storageMock = {
```

Comment:
Discussed with @shuse2 already, I don't think recreating mocks every time is right approach, when we have `jest.clearAllMocks` available.

Reply:
If that would have been the right approach to re-create mocks before each test then there should be a `jest.clearAllMocks` function at all.

Reply:
If we are creating something that needs to be cleared in `each` test, I think right place to define them in beforeEach hooks. But anyway, that's ok. Just then instead of having the call, we should later replace it by setting `clearMocks: true` in jest config.